security-analysis/security-analysis-api/src/main/java/com/powsybl/security/SecurityAnalysisImpl.java

```diff
@@ -88,48 +89,76 @@ public boolean removeInterceptor(SecurityAnalysisInterceptor interceptor) {
 
                         List<Contingency> contingencies = contingenciesProvider.getContingencies(network);
 
-                        futures = new CompletableFuture[contingencies.size()];
-
                         String hash = UUID.randomUUID().toString();
-                        for (int i = 0; i < contingencies.size(); i++) {
-                            Contingency contingency = contingencies.get(i);
-
-                            String postContStateId = hash + "_" + contingency.getId();
-
-                            // run one loadflow per contingency
-                            futures[i] = CompletableFuture
-                                    .supplyAsync(() -> {
-                                        network.getVariantManager().cloneVariant(VariantManagerConstants.INITIAL_VARIANT_ID, postContStateId);
-                                        network.getVariantManager().setWorkingVariant(postContStateId);
-
-                                        // apply the contingency on the network
-                                        contingency.toTask().modify(network, computationManager);
-
-                                        return null;
-                                    }, computationManager.getExecutor())
-                                    .thenComposeAsync(aVoid -> LoadFlow.runAsync(network, postContStateId, computationManager, postContParameters), computationManager.getExecutor())
-                                    .handleAsync((lfResult, throwable) -> {
-                                        network.getVariantManager().setWorkingVariant(postContStateId);
-
-                                        resultBuilder.contingency(contingency)
-                                                .setComputationOk(lfResult.isOk());
-                                        violationDetector.checkAll(contingency, network, resultBuilder::addViolation);
-                                        resultBuilder.endContingency();
-                                        network.getVariantManager().removeVariant(postContStateId);
-
-                                        return null;
-                                    }, computationManager.getExecutor());
+                        int workerCount = Math.min(computationManager.getResourcesStatus().getAvailableCores(), contingencies.size());
+                        BlockingQueue<Integer> queue = new ArrayBlockingQueue<>(workerCount);
+                        for (int i = 0; i < workerCount; i++) {
+                            String postContStateId = hash + "_" + i;
+                            network.getVariantManager().cloneVariant(VariantManagerConstants.INITIAL_VARIANT_ID, postContStateId);
```

Comment:
Here we should clone the `workingStateId` used for pre contingency computation.

Reply:
Yes, I didn't change the original behavior in the first proposal because it was not related. It's now included